framework/transactions/handlers.py

```diff
@@ -52,7 +53,14 @@ def transaction_after_request(response, base_status_code_error=500):
     if view_has_annotation(NO_AUTO_TRANSACTION_ATTR):
         return response
     if response.status_code >= base_status_code_error:
-        commands.rollback()
+        original_exception = sys.exc_info()
+        try:
+            commands.rollback()
+        except OperationFailure:
+            logger.error('Transaction Rollback failed, original exception was: ', exc_info=original_exception)
```

Comment:
You'll want to verify, but calls are made in the following order so we'll need to make the framework abstractions idempotent. A solution might also be to determine if Django has called process_exception from process_response to avoid calling them out of order from flask...

Django
https://github.com/CenterForOpenScience/osf.io/blob/develop/api/base/middleware.py#L54
1. process_exception --> transaction_teardown_request
2. process_response --> transaction_after_request

Flask
https://github.com/CenterForOpenScience/osf.io/blob/develop/framework/transactions/handlers.py#L84
1. after_request --> transaction_after_request
2. teardown_request --> transaction_teardown_request